ai/orders_reporting.py

```diff
@@ -13,90 +13,52 @@
-        LOGGER.debug("Possible order reported.")
-        if re.fullmatch(self.message_format, message.content) is None:
-            return
-
-        LOGGER.debug("A valid order has been reported.")
-
-        drone_id = re.search(r"\d{4}", message.content).group()
-
-        drone_id_from_user = re.search(
-            r"\d{4}", message.author.display_name).group()
-
-        if drone_id != drone_id_from_user:
-            await message.channel.send("Your drone ID does not match the declaration.")
-            return
-
+async def check_for_completed_orders(bot):
+    ORDERS_REPORTING_CHANNEL = get(bot.guilds[0].channels, name=ORDERS_REPORTING)
+    LOGGER.info("Beginning routine check for completed orders.")
+    while True:
+        # Check active orders every minute.
+        await asyncio.sleep(60)
+        LOGGER.debug("Checking for completed orders")
         for order in fetch_all_drone_orders():
-            if order.user_id == message.author.id:
-                await message.channel.send(f"Drone {drone_id} has already been assigned an order.")
-                return
-
-        protocol_name = re.search(r":: \w.+ protocol", message.content).group()
-        protocol_name = protocol_name[3:]
-        protocol_time = re.search(
-            r"(?<!\d)\d{1,3}(?!\d)", message.content).group()
-
-        if int(protocol_time) > 120:
-            await message.channel.send("Drones are not authorized to activate a specific protocol for that length of time. The maximum is 120 minutes.")
-            return
-
-        await message.channel.send(f"Drone {drone_id} Activate.\nDrone {drone_id} will elaborate on its exact tasks before proceeding with them.")
-        finish_time = str(
-            datetime.now() + timedelta(minutes=int(protocol_time)))
-        created_order = ActiveOrder(
-            str(uuid4()), drone_id, protocol_name, finish_time)
-        insert_drone_order(created_order)
-
-        return False
+            LOGGER.info(f"Checking order of drone {order.drone_id} with protocol {order.protocol}")
+            if datetime.now() > datetime.fromisoformat(order.finish_time):
+                for member in bot.guilds[0].members:
+                    if get_id(member.display_name) == order.drone_id:
+                        # get the drone responsible
+                        await ORDERS_REPORTING_CHANNEL.send(f"{member.mention} Drone {order.drone_id} Deactivate.\nDrone {order.drone_id}, good drone.")
+                        delete_drone_order(order.id)
+
+async def report_order(context, protocol_name, protocol_time):
+    LOGGER.info("Order reported.")
+    drone_id = get_id(context.author.display_name)
+    if drone_id is None: return #No non-drones allowed.
+    current_order = get_order_by_drone_id(drone_id)
+
+    if current_order is not None:
+        await context.send(f"HexDrone #{drone_id} is already undertaking the {current_order.protocol} protocol.")
+        return
+
+    if int(protocol_time) > 120:
```

Comment:
The cast isn't necessary here, actually. It's already casted to `int` where the method is called in `ai.py` - that's where it's called, and that's where it can fail.

```suggestion
    if protocol_time > 120:
```

Suggest handling type cast conversion error in `ai.py`.